llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -3079,7 +3079,19 @@ def : GCNPat<
 >;
 } // End OtherPredicates
 
-let OtherPredicates = [HasTrue16BitInsts] in {
+let True16Predicate = UseRealTrue16Insts in {
+def : GCNPat<
+  (fcanonicalize (f16 (VOP3Mods f16:$src, i32:$src_mods))),
```

Comment:
Title is misleading and the test is mismatched. This is an canonicalize pattern, not fmul

Reply:
Right. I need to update the canonicalize test file.

It seems the canonicalize test file depends on V_CMP t16 and it's not supported yet. I'll move this to draft PR and reenable after it's ready